src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -2,32 +2,193 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using System.Diagnostics.Tracing;
+using System.Reflection;
 using System.Runtime.CompilerServices;
+using System.Text;
 
 namespace System.Diagnostics
 {
     /// <summary>
-    /// Current this EventSource is only here so that the Debugger can inject code using Function evaluation
-    /// We may add actual logging requests as well at some point.   This can be removed if the debugger  no 
-    /// longer needs it.   
+    /// DiagnosticSourceEventSource serves two purposes 
+    /// 
+    ///   1) It allows debuggers to inject code via Function evaluation.  This is the purpose of the
+    ///   BreakPointWithDebuggerFuncEval function in the 'OnEventCommand' method.   Basically even in
+    ///   release code, debuggers can place a breakpoint in this method and and then trigger the
+    ///   DiagnosticSourceEventSource via ETW.  Thus from outside the process you can get a hook that
+    ///   is guarenteed to happen BEFORE any DiangosticSource events (if the process is just starting)
+    ///   or as soon as possible afterward if it is on attach.
+    ///   
+    ///   2) It provides a 'bridge' that allows DiagnosticSource messages to be forwarded to EventListers
+    ///   or ETW.    You can do this by enabling the Microsoft-Diagnostics-DiagnosticSource with the
+    ///   'Events' keyword (for diagnostics purposes, you should also turn on the 'Messages' keyword.  
+    ///   
+    ///   This EventSource defines a EventSource argument called 'FilterAndPayloadSpecs' that defines
+    ///   what DiagnsoticSources to enable and what parts of the payload to serialize into the key-value
+    ///   list that will be forwarded to the EventSource.    If it is empty, all serializable parts of
+    ///   every DiagnosticSource event will be forwarded (this is NOT recommended for monitoring but 
+    ///   can be useful for discover).
+    ///   
+    ///   The FilterAndPayloadSpecs is one long string with a the following structures
+    ///   
+    ///   * It is a newline separated list of FILTER_AND_PAYLOAD_SPEC
```

Comment:
The thing I like about newlines is it breaks up what might otherwise be a very long specification when viewed with normal editing tools.   The code will accept both \n and \r\n as the separator, so should work well on both Windows and Linux.    I am open to other options, but I like newline because it does break it up.